### PR TITLE
[PIPDEP-1080] migrate to github action

### PR DESCRIPTION
**Pull request by erwan-b**

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+      - name: Checkout
+        uses: actions/checkout@v2
+      - run: echo "keepcache=1" >> /etc/dnf/dnf.conf
+      - run: ls -la /var/cache/dnf
+      - name: Cache deps
+        uses: actions/cache@v2
+        with:
+          path: /var/cache/dnf
+          key: centos-dnf-cache
+      - run: ls -la /var/cache/dnf
```
Comment (lavigne958) [inline .github/workflows/test.yaml]: is this necessary ? or was it to debug the ci job ?

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      - run: dnf install -y epel-release dnf-plugins-core
+      - run: dnf config-manager --set-enabled PowerTools
+      - run: dnf install -y autoconf automake file gcc glibc-all-langpacks libtool make wget yasm zlib-devel git sudo
+      - name: deps
+        run: |
+          curl -s https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz | \
+          tar -C /usr/local -xzf - && \
+          ln -snf /usr/local/go/bin/go /usr/bin
+          wget -q -O - https://github.com/golangci/golangci-lint/releases/download/v${GOLINT_VERSION}/golangci-lint-${GOLINT_VERSION}-linux-amd64.tar.gz | \
+          tar xvz -C /usr/bin --strip-components=1 -f -
```
Comment (lavigne958) [inline .github/workflows/test.yaml]: This is worth putting i a dedicated shell script.

**Review by lavigne958 — APPROVED** [review]

**Review by bertranddemiddelaer — APPROVED** [review]

**Comment by erwan-b** [comment]
ping

**Pull request merged**